### ISSUE TITLE
DI

### ISSUE THREAD

**Issue by jorbor** [olgakl/OlgasInventorySystem#3]
The readme refers to using dependency injection. What specifically is this in reference to? The `ProductsController(IProductRepository)` constructor?

**Comment by olgakl** [comment]
Yes, dependency inversion (I could have used Unity for dependency
injection). If later you introduce a different type of a data layer, your
new classes can simply implement the IProductRepository interface and you
would not have to change any of the other classes. This makes classes
loosely coupled, and easier to test as well. In tests it is good to have
mock/fake objects for items that might vary such as data in a database/real
repository. This way you have control of what you are testing and tests
that test one functionality do not depend on database
availability/connectivity.
